Microsoft.VisualStudio.DSC/Microsoft.VisualStudio.DSC.psm1

```diff
@@ -240,6 +240,8 @@ function Invoke-VsInstaller
 
     $installer = Start-Process -FilePath (Get-VsInstallerPath) -ArgumentList $Arguments -PassThru
     $installer.WaitForExit();
+    # Wait for all processes in case there is an installer update
+    Wait-Process -name setup;
```

Comment:
Let's make this a bit more robust - there are a lot of processes around called `setup.exe`. How about something like:

```pwsh
$basePath = Join-Path -Path "${env:ProgramFiles(x86)}" -ChildPath "Microsoft Visual Studio"
$activeInstallerProcessIds = Get-Process Setup | Where-Object { $_.Path -like "$basePath*" } | Select-Object -ExpandProperty Id
Wait-Process $activeInstallerProcessIds -Timeout 3600
```

Bonus points if you break out the base path builder into a function and use that also in `Get-VsInstallerPath` and `Get-VsWherePath`